nixos/modules/system/boot/stage-2-init.sh

```diff
@@ -105,7 +105,7 @@ fi
 
 # Required by the activation script
 install -m 0755 -d /etc
-if [ -d "/etc/nixos" ]; then
+if [ ! -h "/etc/nixos" ]; then
```

Comment:
Maybe only if it doesn't exist? `! -e` depending on if the permission setting is important